checkov/terraform/graph_builder/foreach_handler.py

```diff
@@ -23,12 +25,19 @@ def handle_foreach_rendering(self, foreach_blocks: dict[str, list[int]]) -> None
         self._handle_foreach_rendering_for_resource(foreach_blocks.get(BlockType.RESOURCE))
 
     def _handle_foreach_rendering_for_resource(self, resources_blocks: list[int]) -> None:
+        block_index_to_statement = self._get_statements(resources_blocks)
+        # new_resources = self._create_new_foreach_resources(block_index_to_statement)
+
+    def _get_statements(self, resources_blocks: list[int]) -> dict[int, Optional[list[str] | dict[str, Any] | int]]:
+        block_index_to_statement: dict[int, Optional[list[str] | dict[str, Any] | int]] = {}
         for block_index in resources_blocks:
             foreach_statement = self._get_foreach_statement(block_index)
+            block_index_to_statement[block_index] = foreach_statement
             # empty foreach_statement -> leave the main resource
-            if foreach_statement is None:
-                continue
-            # new_resources = self._create_new_foreach_resources(i, foreach_statement)
+        blocks_to_render = [block_idx for block_idx, statement in block_index_to_statement.items() if statement is None]
```

Comment:
just for understanding - why do we want to render only if `statement is None`? 
Intuitively I would assume it should be the opposite